core/src/main/java/com/datastax/oss/driver/api/core/data/CqlVector.java

```diff
@@ -15,26 +15,78 @@
  */
 package com.datastax.oss.driver.api.core.data;
 
-import com.datastax.oss.driver.shaded.guava.common.base.Joiner;
+import com.datastax.oss.driver.api.core.type.codec.TypeCodec;
+import com.datastax.oss.driver.shaded.guava.common.base.Splitter;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
+import com.datastax.oss.driver.shaded.guava.common.collect.Iterables;
 import com.datastax.oss.driver.shaded.guava.common.collect.Iterators;
+import com.datastax.oss.driver.shaded.guava.common.collect.Streams;
+import edu.umd.cs.findbugs.annotations.NonNull;
 import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
 
-/** An n-dimensional vector defined in CQL */
-public class CqlVector<T> {
+/**
+ * An immutable n-dimensional vector representation of the CQL vector type.
+ *
+ * <p>Instances may be created iteratively using {@link CqlVector.Builder} or in a single method
+ * call via {@link #newInstance(Object[])} or {@link #newInstance(Collection)}.
+ */
+public class CqlVector<T> implements Iterable<T> {
```

Comment:
There is still a critical need to have simple access to the underlying values. The iterative accessor is simply not sufficient for our application's needs.

Reply:
Looking at the pros/cons, my understanding is:

* Iterable access provides a generic approach and could be used where lazy evaluation and large datasets are known. 
* List<T> commonly used interface, easy way to work with collections, random access, specific list operations.  This also allows for getting an idea of the size.

Since our full set of vector search use cases are somewhat vague at this point, +1 for having both options.

Reply:
I attempted to address this by adding an export(Collection) method which dumps the vectors content into a collection type of your choosing.  This is significantly more flexible than exporting a single collection type; whatever type we export may have properties (list vs. set vs. queue, immutable vs. immutable, linked lists vs. array lists and so on) which almost certainly will not apply in even a majority of situations.  This approach allows API users to create the list in whatever way they see fit (and even perform some initial operations on said list if they choose to do so) and then hand it over to our vector for supplying the data.

It's my hope that this addresses your immediate concern @jshook while also maximizing our flexibility for the many Java driver users who will have use cases with different parameters.  If I outright missed the boat here and didn't understand something please let me know.

Reply:
This is an ergonomic improvement, and quite flexible. However, it doesn't improve the situation around the heap allocation acrobatics for large (assume 2K nominal) vectors. I'm working up some microbenches to quantify this better. It seems worth it at least for something that, once settled, will be how CqlVector works for a long time.